gitpod-io/gitpod
# [metrics] Add metrics for unary and streaming gRPC server

This PR adds metrics for all gRPC servers using https://github.com/grpc-ecosystem/go-grpc-prometheus.

Fixes https://github.com/gitpod-io/gitpod/issues/3090

### How to test
1. Start a workspace to produce some traffic in the system
2. Check the metrics of ws-manager, ws-daemon and image-builder. You should see `grpc_server_started_total`, `grpc_server_sent_total`, `grpc_server_msg_received_total` and `grpc_server_handling_seconds_` metrics.
